lib/propmangle.js

```diff
@@ -159,7 +162,7 @@ function mangle_properties(ast, options) {
         if (options.only_cache) {
             return cache.props.has(name);
         }
-        if (/^[0-9.]+$/.test(name)) return false;
+        if (/^-?[0-9]+(\.[0-9]+)?(e[+-][0-9]+)?$/.test(name)) return false;
```

Comment:
> Wait a sec... if it's raw then what do you receive as a property for ({ 0x25: 1 }) in  can_mangle()  /  should_mangle()  ?

Ah, you finally got to the same revelation I had at the start of this PR.

The _number_ `37` is being passed to `can_mangle()`, but `RegExp.test()` will turn its parameter into a string before acting on it, negating the potential nasty surprises :wink: 

This thing is black magic.